crates/uv-workspace/src/workspace.rs

```diff
@@ -0,0 +1,69 @@
+use std::path::{Path, PathBuf};
+
+use uv_fs::Simplified;
+
+use crate::{Options, PyProjectToml};
+
+/// Represents a project workspace that contains a set of options and a root path.
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct Workspace {
+    options: Options,
+    root: PathBuf,
+}
+
+impl Workspace {
+    /// Find the [`Workspace`] for the given path.
+    ///
+    /// The search starts at the given path and goes up the directory tree until a workspace is
+    /// found.
+    pub fn find(path: impl AsRef<Path>) -> Result<Option<Self>, WorkspaceError> {
+        for ancestor in path.as_ref().ancestors() {
+            if let Some(options) = read_options(ancestor)? {
+                return Ok(Some(Self {
+                    options,
+                    root: ancestor.to_path_buf(),
+                }));
```

Comment:
Does this mean that we'll fail on _any_ uv invocation if there's an unparsable `pyproject.toml` file in an ancestor directory?

This behavior kind of drives me crazy in Ruff.

Reply:
I'd rather log a warning and return `None`. I think failing on an unparsable `uv.toml` makes more sense to me though.

Reply:
Yes it does.

Reply:
Fixed this. We show a user-facing warning on `pyproject.toml` but error on `uv.toml`.